geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/service/GeoMesaDataStoreService.scala

```diff
@@ -39,19 +39,23 @@ class GeoMesaDataStoreService[T <: DataStoreFactorySpi: ClassTag](descriptors: S
 
   override def loadDataStore: DataStore = synchronized {
     if (store == null) {
-      store = tryGetDataStore(params).get
+      store = newDataStore()
     }
     store
   }
 
+  override def newDataStore(): DataStore = synchronized {
+    tryGetDataStore(params).get
+  }
```

Comment:
```suggestion
  override def newDataStore(): DataStore = tryGetDataStore(params).get
```
this doesn't need to be synchronized i don't think...

Reply:
the synchronization is just needed for managing the state of the singleton `store`